Dockerfile

```diff
@@ -1,18 +1,31 @@
-# ---- Base image ----
-FROM hmctspublic.azurecr.io/base/node:20-alpine as base
+FROM hmctspublic.azurecr.io/base/node:20-alpine as build
+
+# Run all commands as hmcts user
 USER root
 RUN corepack enable
-COPY --chown=hmcts:hmcts . .
 USER hmcts
 
-# ---- Build image ----
+WORKDIR /opt/app
+# Copy necessary files
+COPY --chown=hmcts:hmcts package.json yarn.lock webpack.config.js tsconfig.json .eslintrc.js .eslintignore .yarnrc.yml ./
+
+# Install dependencies
+# Copy .yarn directory
+COPY --chown=hmcts:hmcts .yarn/ .yarn/
+COPY --chown=hmcts:hmcts config/ ./config/
+# Verify contents
+RUN ls -l /opt/app/.yarn
+RUN PUPPETEER_SKIP_DOWNLOAD=true yarn install
 
-FROM base as build
-RUN PUPPETEER_SKIP_DOWNLOAD=true yarn install && yarn build:prod
+# Copy application files
+COPY --chown=hmcts:hmcts src/main/ ./src/main/
+COPY --chown=hmcts:hmcts webpack/ ./webpack/
```

Comment:
I think this should just be `COPY --chown... . .` so that any remaining files get copied, otherwise this is a maintenance pain, dockerignore should take care of only ensuring wanted files get copied